app/models/foreman_docker/service/actions/container/pull.rb

```diff
@@ -0,0 +1,28 @@
+module ForemanDocker
+  module Service
+    module Actions
+      module Container
+        class Pull < ::Actions::EntryAction
+          def plan(container)
+            action_subject(container)
+            plan_self(:container => container.id)
+          end
+
+          def run
+            container = ::Container.find(input[:container])
+            container.compute_resource.create_image(:fromImage => container.repository_pull_url)
```

Comment:
After discussion with @pitr-ch offline, here are some comments to this approach:

1. the active record calls should be kept outside of the Thread.new block, as every such operation blocks the connection from the ActiveRecord pool until the thread finishes: the question is whether `container.compute_resource.create_image(:fromImage => container.repository_pull_url)`
2. in connection to 1., ideally, we should have a mechanism, how to enforce the block of the code in the Thread.new to not be able to get the ActiveRecord connection for the very same reason. Not necessary needed, but helpful
3. there should be an ensure block sending an event to suspended_action, to make sure the suspeded action wakes up. I would presonally recommend to do `resuce Exception => e` and reraise it after sending the event to the suspended action